test/framework/src/main/java/org/elasticsearch/test/AbstractQueryTestCase.java

```diff
@@ -153,6 +159,10 @@
     private static String[] currentTypes;
     private static String[] randomTypes;
 
+    /**
+     * used to check warning headers of the deprecation logger
+     */
+    protected ThreadContext threadContext;
```

Comment:
does this need to be protected? I think it is safer if it is private and the only way to access it is via checkWarningHeaders